examples/rust/files/chunk_sender/src/main.rs

```diff
@@ -0,0 +1,654 @@
+#![allow(unused_mut, dead_code)]
+#![allow(unused_macro_rules)]
+
+use std::{
+    env,
+    net::{Ipv4Addr, SocketAddr},
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+use anyhow::{bail, Context as _, Result};
+use aranya_client::{
+    aqc::{AqcPeerChannel, AqcBidiChannel}, client::Client, Error, QuicSyncConfig, SyncPeerConfig, TeamConfig,
+};
+use aranya_daemon_api::{text, ChanOp, DeviceId, KeyBundle, NetIdentifier, Role};
+use aranya_util::Addr;
+use backon::{ExponentialBuilder, Retryable};
+use buggy::BugExt;
+use bytes::Bytes;
+use tempfile::TempDir;
+use tokio::{
+    fs,
+    process::{Child, Command},
+    time::sleep,
+    io::AsyncReadExt,
+};
+use tokio::fs::File;
+use tracing::{debug, info, Metadata};
+use tracing_subscriber::{
+    layer::{Context, Filter},
+    prelude::*,
+    EnvFilter,
+};
+use serde::{Serialize, Deserialize};
+use serde_cbor;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct FileChunk {
+    filename: String,
+    chunk_index: u64,
+    chunk_size: usize,
+    total_chunks: u64,
```

Comment:
This only needs to be sent with the first chunk.
```suggestion
    total_chunks: Option<u64>,
```
Or instead of defining optional fields, different types for the first file chunk and the rest of the file chunks could be defined. E.g. `FirstFileChunk` and `FileChunk`.